aten/src/ATen/native/Convolution.cpp

```diff
@@ -207,6 +218,43 @@ static void check_input_shape_forward(const at::Tensor& input,
              "Given weight of size ", weight.sizes(),
              ", expected bias to be 1-dimensional with ", weight.size(0), " elements",
              ", but got bias of size ", bias.sizes(), " instead");
+
+    for (int i = 2; i < k; ++i) {
+      input_shape.push_back(input.size(i) + 2 * padding[i-2]);
+      kernel_shape.push_back(weight.size(i));
+      if (input_shape.back() < kernel_shape.back()) {
+        kernel_size_correct = false;
+      }
+
+      // TODO: Should output padding be considered while computing output_shape?
+      output_shape.push_back(div_rtn<int64_t>(input_shape.back() - kernel_shape.back(), stride[i-2]) + 1);
```

Comment:
Correct me if I am wrong, I don't think we support negative stride on convolutional layers. If so, can you add an `AT_CHECK` for that? And if so, do you still need the `output_shape` check?


```python
>>> c = nn.Conv1d(3, 2, kernel_size=7, stride=1)                                                                                                                      
>>> c(torch.zeros(2, 3, 10))                                                                                                                                          
tensor([[[-0.1358, -0.1358, -0.1358, -0.1358],
         [-0.0659, -0.0659, -0.0659, -0.0659]],

        [[-0.1358, -0.1358, -0.1358, -0.1358],
         [-0.0659, -0.0659, -0.0659, -0.0659]]], grad_fn=<SqueezeBackward1>)
>>> c = nn.Conv1d(3, 2, kernel_size=7, stride=-1)
>>> c(torch.zeros(2, 3, 10))                                                                                                                                          
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/shenli/project/pytorch/torch/nn/modules/module.py", line 491, in __call__                                                                               
    result = self.forward(*input, **kwargs)
  File "/home/shenli/project/pytorch/torch/nn/modules/conv.py", line 188, in forward                                                                                  
    self.padding, self.dilation, self.groups)
RuntimeError: Trying to create tensor with negative dimension -2: [2, 2, 1, -2]
>>> c = nn.Conv1d(3, 2, kernel_size=7, stride=-3)
>>> c(torch.zeros(2, 3, 10))                                                                                                                                          
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/shenli/project/pytorch/torch/nn/modules/module.py", line 491, in __call__                                                                               
    result = self.forward(*input, **kwargs)
  File "/home/shenli/project/pytorch/torch/nn/modules/conv.py", line 188, in forward                                                                                  
    self.padding, self.dilation, self.groups)
```

Reply:
we don't.